### PR TITLE
update puma configs for K8s deployments

### PR DESCRIPTION
**Pull request by camallen**
scale via k8s pods and reduce the CPU / RAM load in the web server container

As per docs here, https://github.com/puma/puma#thread-pool

THIS needs to be tested locally, i haven't done this yet. `docker-compose build` then `docker-compose up` would do it for non production setup

### TODO
- [x] Look at how the puma restart script will work, https://github.com/zooniverse/Panoptes/blob/5cbaf621f99799880112e7304e9b4016ff78343d/scripts/docker/restart_puma.sh#L5 docs here https://github.com/puma/puma/blob/master/docs/restart.md

# Review checklist

- [ ] First, the most important one: is this PR small enough that you can actually review it? Feel free to just reject a branch if the changes are hard to review due to the length of the diff.
- [ ] If there are any migrations, will they the previous version of the app work correctly after they've been run (e.g. the don't remove columns still known about by ActiveRecord).
- [ ] If anything changed with regards to the public API, are those changes also documented in the `apiary.apib` file?
- [ ] Are all the changes covered by tests? Think about any possible edge cases that might be left untested.

### CODE REVIEW

File: config/puma.rb [hunk 1]
```diff
@@ -9,20 +9,19 @@
...
 state_path "#{app_path}/tmp/pids/puma.state"
 
-if rails_env == "production"
-  stdout_redirect "#{app_path}/log/production.log", "#{app_path}/log/production_err.log", true
-end
-
 bind "tcp://0.0.0.0:#{port}"
 
-# === Cluster mode ===
+threads_count = ENV.fetch("RAILS_MAX_THREADS") { 2 }.to_i
```
[the remark this answers is not in the document]
Comment (camallen) [reply config/puma.rb:14]:
```suggestion
threads_count = ENV.fetch('RAILS_MAX_THREADS', 2).to_i
```

**Comment by adammcmaster** [comment]
Regarding the puma restart script, is that only used when we want to restart/roll the app (e.g. after DB failover)? If so, Kubernetes 1.15 has a [`kubectl rollout restart`](https://kubernetes.io/docs/reference/generated/kubectl/kubectl-commands#-em-restart-em-) feature we can use to restart the pods instead, so we can try that once we upgrade (we were waiting for PH TESS to finish its current data before doing that, but that seems to be finished today).

**Comment by adammcmaster** [comment]
Noting that #3316 should be reverted once the port is updated or once Nginx is added.

**Comment by camallen** [comment]
> Could you also change the default non-dev port from 81 to 80? (Or alternatively, should we be running an Nginx container in the pod to serve assets?)

We will need a nginx reverse proxy pod to serve the static assets, hence the broken assets on https://panoptes-staging.azure.zooniverse.org/

> kubectl rollout restart

Neat - that's very useful, especially for the schema migrations context. I'd still like the puma restart script in the repo (local testing reload code, etc) but it's low priority if we use this after migrations.

**Pull request merged**